internal/auth0/organization/resource_member_role_test.go

```diff
@@ -0,0 +1,113 @@
+	depends_on = [ auth0_organization_member.member ]
+
+	organization_id = auth0_organization.org.id
+	user_id         = auth0_user.user.id
+
+	roles = []
+}
+`
+
+func TestAccOrganizationMemberRole(t *testing.T) {
+	testName := strings.ToLower(t.Name())
+
+	acctest.Test(t, resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ParseTestName(testAccUpdateOrganizationMemberWithOneRoleAssigned, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role1", "user_id"),
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role1", "organization_id"),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role1", "role_name", fmt.Sprintf("Test Reader - %s", testName)),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role1", "role_description", ""),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccUpdateOrganizationMemberWithTwoRolesAssigned, testName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role1", "user_id"),
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role1", "organization_id"),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role1", "role_name", fmt.Sprintf("Test Reader - %s", testName)),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role1", "role_description", ""),
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role2", "user_id"),
+					resource.TestCheckResourceAttrSet("auth0_organization_member_role.role2", "organization_id"),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role2", "role_name", fmt.Sprintf("Test Writer - %s", testName)),
+					resource.TestCheckResourceAttr("auth0_organization_member_role.role2", "role_description", ""),
+				),
+			},
+			{
+				Config: acctest.ParseTestName(testAccGivenTwoRolesAUserAnOrganizationAndAnOrganizationMember, testName),
+			},
+			{
+				Config:       acctest.ParseTestName(testAccRemoveAssignedRolesFromOrganizationMember, testName),
+				ResourceName: "auth0_organization_member_roles.roles",
+				ImportState:  true,
+				ImportStateIdFunc: func(state *terraform.State) (string, error) {
+					organizationID, err := acctest.ExtractResourceAttributeFromState(state, "auth0_organization.org", "id")
+					if err != nil {
+						return "", err
+					}
+
+					userID, err := acctest.ExtractResourceAttributeFromState(state, "auth0_user.user", "id")
+					if err != nil {
+						return "", err
+					}
+
+					return organizationID + ":" + userID, nil
+				},
+				ImportStatePersist: true,
+			},
+			{
+				Config: acctest.ParseTestName(testAccRemoveAssignedRolesFromOrganizationMember, testName),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectEmptyPlan(),
+					},
+				},
+			},
+		},
+	})
+}
```

Comment:
I _think_ I get the gist of it? Perform a delete by removing the 1:1 and then tracking diff changes with the 1:many and asserting that there are no roles existing. Am I correct in my understanding?

Reply:
Yes.